pkg/upgrade/upgrader/jiva_upgrade.go

```diff
@@ -117,35 +118,67 @@ func getControllerPatchDetails(d *appsv1.Deployment) (
 	return patchDetails, nil
 }
 
-func getReplica(replicaLabel, namespace string) (*replicaDetails, error) {
-	replicaObj := &replicaDetails{}
-	deployObj, err := getDeployment(replicaLabel, namespace)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to get replica deployment")
-	}
-	if deployObj.Name == "" {
-		return nil, errors.Errorf("missing deployment name for replica")
-	}
-	replicaObj.name = deployObj.Name
-	version, err := getOpenEBSVersion(deployObj)
+func getPatchDetailsForDeploy(pvName string, deployObj *appsv1.Deployment) (*replicaPatchDetails, error) {
+	patchDetails, err := getReplicaPatchDetails(deployObj)
 	if err != nil {
 		return nil, err
 	}
-	if (version != currentVersion) && (version != upgradeVersion) {
-		return nil, errors.Errorf(
-			"replica version %s is neither %s nor %s\n",
-			version,
-			currentVersion,
-			upgradeVersion,
-		)
+	patchDetails.UpgradeVersion = upgradeVersion
+	patchDetails.PVName = pvName
+	return patchDetails, nil
+}
+
+func getReplica(pvName, replicaLabel, namespace, openebsNamespace string) (*replicaDetails, error) {
+	replicaObj := &replicaDetails{
+		replicas: map[string]string{},
 	}
-	replicaObj.version = version
-	patchDetails, err := getReplicaPatchDetails(deployObj)
+	var err error
+	// check if old replica is present for currentVersion < 1.9.0
+	// if present then migration is not complete and store the old
+	// replica details
+	if currentVersion < "1.9.0" {
```

Comment:
Above comments are good..
add few more like
'replicaObj.name and replicaObj.version would be empty if old replica got deleted as part of upgrade.
So, later on code uses replicaObj.name to perform replica related migration.'